File: algorithm_python/Algorithm_Session/Day2/boj_1541.py
# 잃어버린 괄호

# Input ex) 10+55-50+40-30+10    =>    최솟값: 10+55-(50+40)-(30+10)

a = input().split("-")      # a = ["10+55", "50+40", "30+10"]
plus = a[0]                 # plus = "10+55"          
minus = a[1:]               # minus = ["50+40", "30+10"]
ans = 0

for p in plus.split("+"):   # p <- ["10", "55"]
    ans += int(p)

for m in minus:                 
    for v in m.split("+"):      # i = 0, v <- ["50", "40"]
        ans -= int(v)           # i = 1, v <- ["30", "10"]  

print(ans)

File: algorithm_java/Day2/boj_1541.java
package Day2;

import java.util.Scanner;

public class boj_1541 {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String s = scan.nextLine();

        String a[] = s.split("-");
        int ans = 0;

        for (int i = 0; i < a.length; i++) {
            String b[] = a[i].split("\\+");
            if (i == 0)
                for (int j = 0; j < b.length; j++)
                    ans += Integer.parseInt(b[j]);
            else
                for (int j = 0; j < b.length; j++)
                    ans -= Integer.parseInt(b[j]);
        }

        System.out.println(ans);
        scan.close();
    }
}
